safe_transaction_service/history/tests/test_views.py

```diff
@@ -1091,6 +1091,17 @@ def test_delete_safe_delegate(self):
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
         self.assertEqual(SafeContractDelegate.objects.count(), 1)
 
+        # Check if a delegate can delete itself
+        SafeContractDelegate.objects.all().delete()
+        delegate_account = Account().create()
+        SafeContractDelegateFactory(safe_contract=safe_contract, delegate=delegate_account.address)
+        hash_to_sign = DelegateSignatureHelper.calculate_hash(delegate_account.address)
+        data['signature'] = delegate_account.signHash(hash_to_sign)['signature'].hex()
+        response = self.client.delete(reverse('v1:history:safe-delegate', args=(safe_address, delegate_account.address)),
```

Comment:
I am not sure if this already behaves like so, but if we allow to delete several delagators, should we make this a transactional method?

Reply:
What do you mean?